readest: 0.9.43 -> 0.9.51; disable telemetry

ddogfoodd opened the pull request:
Changelog: https://github.com/readest/readest/releases/tag/v0.9.51
Diff: https://github.com/readest/readest/compare/v0.9.43...v0.9.51

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

ddogfoodd committed:
readest: 0.9.43 -> 0.9.49

Changelog: https://github.com/readest/readest/releases/tag/v0.9.49
Diff: https://github.com/readest/readest/compare/v0.9.43...v0.9.49

eljamm commented:
> Is there a way in which we can make use of the [new telemetry opt-out](https://github.com/readest/readest/commit/68c3c45da37d3d31e05bfe37613235d6b86977cd)? I think it would be great if we could package with opt-out by default -> turn it into opt-in.

I agree, and I think we'd only have to [patch this line](https://github.com/readest/readest/commit/68c3c45da37d3d31e05bfe37613235d6b86977cd#diff-1eddf341eaa6329a617ca3cba384ffacb59619a153c02a95bab0c19fd070b6c2R44) to `false`.

ddogfoodd committed:
readest: disable telemetry

add patch to disable telemetry

eljamm reviewed (approved):
Tested this and the only issue I found is that I couldn't invert the image for a specific book (others were fine), but other than that things are looking good and telemetry is off by default.

**PS:** the new bilingual features in this release are pretty neat :)

ddogfoodd committed:
readest: 0.9.49 -> 0.9.50

Changelog: https://github.com/readest/readest/releases/tag/v0.9.50
Diff: https://github.com/readest/readest/compare/v0.9.49...v0.9.50

donovanglover commented:
AFAICT `optOutTelemetry()` only gets ran when toggling telemetry in the UI so [`TELEMETRY_OPT_OUT_KEY`](https://github.com/readest/readest/commit/68c3c45da37d3d31e05bfe37613235d6b86977cd#diff-57e7b1f54041418743762e4822ef9db6b345a860900ff8ea55df09b2d0b0b115R3) would also have to be set to `'true'`.

eljamm commented:
> AFAICT `optOutTelemetry()` only gets ran when toggling telemetry in the UI so [`TELEMETRY_OPT_OUT_KEY`](https://github.com/readest/readest/commit/68c3c45da37d3d31e05bfe37613235d6b86977cd#diff-57e7b1f54041418743762e4822ef9db6b345a860900ff8ea55df09b2d0b0b115R3) would also have to be set to `'true'`.

If I'm not mistaken, setting `telemetryEnabled` to false is the equivalent of turning it off in the UI.

```shellSession
$ rm ~/.config/com.bilingify.readest/settings.json
$ nix-build -A readest && ./result/bin/readest 
$ cat ~/.config/com.bilingify.readest/settings.json | jq '.telemetryEnabled'
false
```

Which can also be observed from the unticked mark in the UI menu.

> Else we could just set `TELEMETRY_OPT_OUT_KEY` additionally, doesn't hurt to double opt-out, right?

That said, I suppose we can do this as well for good measure.

Btw, a new version is out: https://github.com/readest/readest/releases/tag/v0.9.51

ddogfoodd committed:
readest: 0.9.50 -> 0.9.51

Changelog: https://github.com/readest/readest/releases/tag/v0.9.51
Diff: https://github.com/readest/readest/compare/v0.9.50...v0.9.51

eljamm commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 411465`

Logs: https://github.com/eljamm/nixpkgs-review-gha/actions/runs/15366074416


---
### `x86_64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>readest</li>
  </ul>
</details>


---
### `aarch64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>readest</li>
  </ul>
</details>

eljamm reviewed (approved):
Re-tested briefly and it seems to be working fine.
